src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -873,3 +906,154 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
 
   return TRUE;
 }
+
+/**
+ * ostree_repo_static_delta_is_signed:
+ * @self: Repo
+ * @delta_id: delta path
+ * @out_value: (out) (element-type utf8) (transfer container): location to store
+ *            the list of signature type strings.
+ * @error: Error
+ *
+ * Verify if the static-delta has signature(s).
+ * If the option is not set, or if an error is returned, @out_value will be set
+ * to %NULL.
+ *
+ * Returns: TRUE if static data has signatures, FALSE otherwise.
+ *
+ * Since: 2020.1
+ */
+gboolean
+ostree_repo_static_delta_is_signed (OstreeRepo       *self,
```

Comment:
One property of GVariant is it will never fail - you can take any arbitrary bytestream and interpret it as GVariant of any given type and it will always return *something*.

What's going on here is pretty weird because we're basically scraping the first N bytes for any strings matching `ostree.sign.`.  This seems potentially fragile - what if a non-signed delta *happened* to contain those bytes somehow (e.g. we're updating ostree itself and the target binary embedded in the delta payload contains it).

Upon reflection...it might be cleaner to define a magic header for these signed deltas.  For example we could require the first bytes be `OSTSGNDT` (short for "ostree signed delta") and then the data after *that* is GVariant.  The reason for 8 bytes is alignment so we can `mmap()`.  WDYT?

(Also since I've now given bad advice on this PR I am happy to run with the ball a bit and update things to do that)

Reply:
Or alternatively, we could require signed deltas to have a filename suffix like `.ostsd` or so.   This would seem OK for the use case of "copy delta to USB stick", but I guess is more problematic for storing them in a repository.

Reply:
I pushed a new version with the magic header.
I also changed the order of signatures and delta variant, i.e. `taya{sv}` where:
- t: magic header `OSTSGNDT`
- ay: existing delta variant
- a{sv}: signatures

With this order change GLib takes care of alignment and it removes the dependency on Glib2 >= 2.60